### ISSUE TITLE
Insufficient contrast for WCAG in button hover states

### ISSUE THREAD

**Issue by tkoleary** [massgov/mayflower#869]
-Mayflower React
-Component: ButtonSearch

<img width="139" alt="Screenshot 2020-01-03 12 03 48" src="https://user-images.githubusercontent.com/3894652/71740437-d032a580-2e21-11ea-9c7e-27932a4a388b.png">

In ButtonWithIcon the text stays white on hover, but here it is changing to green which has insufficient contrast. It looks like an oversight, in any event it's an easy fix I'd be happy to contribute.

**Comment by clairesunstudio** [comment]
Good catch! ButtonSearch is a variant of ButtonWithIcon since https://github.com/massgov/mayflower/pull/81. The issue I think is that ButtonSearch should have  its own usages instead of inheriting the ButtonWithIcon usages
- search primary --> https://mayflower.digital.mass.gov/?p=atoms-button-search
  - default: solid background with white content
  - hover state: inverted
- search secondary --> https://mayflower.digital.mass.gov/?p=atoms-button-search-as-secondary
  - default: inverted
  - hover state: solid background with white content

See https://github.com/massgov/mayflower/pull/261 for the button search options and corresponding classes.

Let me know if you would like to [create a pull request](https://github.com/massgov/mayflower/blob/master/CONTRIBUTING.md). This might not be as straight forward as we will have to customize the usages for a variant. So I'm also happy to work on a fix and maybe you can review it.
